src/app.js

```diff
@@ -29,43 +46,107 @@ passport.deserializeUser(function(obj, cb) {
   cb(null, obj);
 });
 
-const app = express();
-app.use(express.static('public'))
+app.use(express.static('public'));
 app.use(require('morgan')('combined'));
 app.use(require('body-parser').urlencoded({ extended: true }));
-app.use(require('express-session')({ secret: 'keyboard cat', resave: true, saveUninitialized: true }));
+app.use(require('express-session')({ secret: 'keyboard cat', resave: false, saveUninitialized: false }));
 
 app.use(passport.initialize());
-app.use(passport.session());
+// app.use(passport.session());
 
-app.get('/', (request, response) => {
-  response.send('index.html');
+app.get('/login/twitter', passport.authenticate('twitter'), (request, response) => {
+  response.send({ hello: 'this is the auth route' });
 });
 
-app.get('/login', (request, response) => {
-  response.sendStatus(200);
-});
+app.get(
+  '/oauth/callback',
+  passport.authenticate('twitter', { failureRedirect: '/login/twitter' }),
+  (request, response) => {
+    const sessionId = request.sessionID;
+    const sessionData = request.sessionStore.sessions[sessionId];
+    const oAuthData = JSON.parse(sessionData)['oauth:twitter'];
 
-app.get('/login/twitter', passport.authenticate('twitter'));
+    const userData = request.user._json;
 
-app.get('/oauth/callback',
-  passport.authenticate('twitter', { failureRedirect: '/login' }),
-  (request, response) => {
-    response.redirect('exp://8q-3ig.kozlowskicd.react-native-spike.exp.direct:80');
+    const savedUserData = {
+      userId: userData.id,
+      userName: userData.name,
+      userScreenName: userData.screen_name,
+      photoLink: request.user.photos[0].value,
+      oAuthToken: request.query.oauth_token,
+      oAuthVerifier: request.query.oauth_verifier,
+      oAuthTokenSecret: oAuthData.oauth_token_secret,
+    };
+
+    // add to database
+
+    // send username, displayname and id to frontend
+
+    // response.send(savedUserData);
+    response.redirect(
+      `exp://exp.host/@melissastock/front-end/?display_name=${savedUserData.userScreenName}&user_name=${
+        savedUserData.userName
+      }&id=${savedUserData.userId}`
+    );
   }
 );
 
 app.get('/dashboard', (request, response) => {
   response.send('Logged in!');
 });
 
+app.get('/test', (request, response) => {
+  return database.query(sql.test)
+    .then(result => {
+      if(result) {
+        response.send(result.rows);
+      }
+      else {
+        response.send('Whoops');
```

Comment:
Why 'Whoops' instead of a 401 or 400?

Reply:
Was there for testing, but it's now changed to send 400 status.